### ISSUE TITLE
Add auto-export like mcchrish/zenbones.nvim?

### ISSUE THREAD

**Issue by rktjmp** [rktjmp/lush.nvim#62]
mcchrish/zenbones.nvim ships with `colors/zenbones.vim` which contains an exported lush theme and `colors/zenbones-lua.vim` for the lush loaded theme.

This seems.... really obvious....

Exporting a theme could (will) impact existing theme files though, and it would make developing themes a bit awkward (`colorsheme my-theme-lua`), but it should probably be a function (auto export) and maybe some affordances made in the template repo for this. 

The export *could* place some in-file markers and only replace inside them, which might let it be a bit more opt in and less violent.

---

Probably:

- `lush-template` still ships with lush-loading `theme.vim`.
- `:Lush export|print|out|save|freeze <fname>` will:
  - Warn user we are making a work-flow destructive change (they will have to update config to load the "live" version")
  - Export the currently loaded theme to `theme.vim` in-between some markers
  - Disable lush-loading in `theme.vim` (probably just put the lush stuff inside the markers)
  - Export a fresh lush-loading theme to `theme-lua|lush|live.vim` (or copy the existing file at step 1)

This fixes a long standing pet-peev of mine where end user lush-themes generally require a runtime dependency (lush).

---

Painpoints:

- light/dark etc themes
  - need option to specify specific files
- themes with optional parts (options to load plugin themes or low/high contrast modes)
  -

**Comment by rktjmp** [comment]
Probably:

- `lush-template` still ships with lush-loading `theme.vim`.
- `:Lush export|print|out <fname>` will:
  - Warn user we are making a work-flow destructive change (they will have to update config to load the "live" version")
  - Export the currently loaded theme to `theme.vim` in-between some markers
  - Disable lush-loading in `theme.vim` (probably just put the lush stuff inside the markers)
  - Export a fresh lush-loading theme to `theme-lua|lush|live.vim` (or copy the existing file at step 1)

This fixes a long standing pet-peev of mine where end user lush-themes generally require a runtime dependency (lush).

**Comment by mcchrish** [comment]
I just built out a very simple template building structure to zenbones.
Example template: https://github.com/mcchrish/zenbones.nvim/blob/main/lua/zenbones/build/alacritty.lua
Build script: https://github.com/mcchrish/zenbones.nvim/blob/main/lua/zenbones/build.lua

I copied a lot of ideas from melange and tokyonight.

The build script is just a wrapper of this simple string interpolation solution http://lua-users.org/wiki/StringInterpolation then write it out to the specified file path.

I got the luxury of having just a light theme so the implementation is quite simple.
